File: p3/p3.js
// The prime factors of 13195 are 5, 7, 13 and 29.

// What is the largest prime factor of the number 600851475143 ?

function is_prime(input){
    var output = true;
    for (var i = 2; i * i < input; i++){
        if (input % i == 0){
            output = false
        }
    }
    return output
}

function big_factor(input){
    var output;
    for (var i = 1; i * i < input; i++){
        if (input % i === 0 && is_prime(i)){
            output = i
        }
    }
    return output
}

console.log(big_factor(600851475143))

File: p3/p3.clj
;; The prime factors of 13195 are 5, 7, 13 and 29.

;; What is the largest prime factor of the number 600851475143 ?


(defn prime? [n]
  (if (even? n)
    false
    (not-any? #(zero? (mod n %))
      (range 3 (int (inc (Math/sqrt n))) 2))))

(defn prime_factor? [a b]
  (and
    (zero? (mod a b))
    (prime? b)))

(def start (bigint 600851475143))

(first
  (filter #(prime_factor? start %)
    (range (int (Math/sqrt start)) 1 -1)))
; 6857

